app/scripts/modules/core/src/instance/instanceSearchResultType.tsx

```diff
@@ -76,6 +79,18 @@ class InstancesSearchResultType extends SearchResultType<IInstanceSearchResult>
     );
   };
 
+  public search(
+    params: ISearchParams,
+    _otherResults: Observable<ISearchResultSet>,
+  ): Observable<ISearchResults<IInstanceSearchResult>> {
+    const { cloudProvider, key, ...otherParams } = params;
+    // Because the instance search is an exact match, we can detect when an aws search is missing 'i-' and prepend it
+    if (cloudProvider === 'aws' && key.match(new RegExp('^[A-Fa-f0-9]{17}$'))) {
+      return super.search({ ...otherParams, key: `i-${key}` }, _otherResults);
```

Comment:
does using `otherParams` here mean that `cloudProvider` will be excluded (vs. the call below that uses `params`)? Is that a concern?

Reply:
Ah, you're right. That's unintentional. Thanks for catching that.